vllm/model_executor/parameter.py

```diff
@@ -328,6 +328,55 @@ def adjust_shard_indexes_for_packing(self, shard_size, shard_offset):
             marlin_tile_size=self.marlin_tile_size)
 
 
+class HQQQweightParameter(PackedvLLMParameter):
```

Comment:
@dsikka @mgoin @LucasWilkinson what do you guys think of inheriting from the PackedParameter to handle the details of HQQ? I think this is a reasonable approach.

Perhaps this should live inside `hqq_marlin.py`

Reply:
@ElizaWszola would like to understand how we're extending this class to see if we can clean-up some of the weight loading logic.

Reply:
@dsikka HQQ qweights have a data format where each row corresponds to 2 groups. I'm first reshaping HQQ qweights (and correspoinding scales/zp) to their actual shapes in qweight multiplication, and then load. Additionally, I store shard offsets and sizes because they are needed for unpacking HQQ from 4-bits to 8-bits (I need to unpack shard-by-shard, so I must know where they are).